receiver/aerospikereceiver/integration_test.go

```diff
@@ -0,0 +1,73 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integration
+// +build integration
+
+package aerospikereceiver_test
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/containertest"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/scrapertest"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/scrapertest/golden"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/aerospikereceiver"
+)
+
+func TestAerospikeIntegration(t *testing.T) {
+	t.Parallel()
+
+	ct := containertest.New(t)
+	container := ct.StartImage("aerospike:ce-5.7.0.17", containertest.WithPortReady(3000))
+
+	f := aerospikereceiver.NewFactory()
+	cfg := f.CreateDefaultConfig().(*aerospikereceiver.Config)
+	cfg.Endpoint = container.AddrForPort(3000)
+	cfg.ScraperControllerSettings.CollectionInterval = 100 * time.Millisecond
+
+	consumer := new(consumertest.MetricsSink)
+	settings := componenttest.NewNopReceiverCreateSettings()
+	receiver, err := f.CreateMetricsReceiver(context.Background(), settings, cfg, consumer)
+	require.NoError(t, err, "failed creating metrics receiver")
+	require.NoError(t, receiver.Start(context.Background(), componenttest.NewNopHost()), "failed starting metrics receiver")
+
+	require.Eventually(t, func() bool {
+		return consumer.DataPointCount() > 0
+	}, 2*time.Minute, 1*time.Second, "failed to receive more than 0 metrics")
+	require.NoError(t, receiver.Shutdown(context.Background()), "failed shutting down metrics receiver")
+
+	actualMetrics := consumer.AllMetrics()[0]
+	expectedFile := filepath.Join("testdata", "integration", "expected.json")
+	expectedMetrics, err := golden.ReadMetrics(expectedFile)
+	require.NoError(t, err, "failed reading expected metrics")
+
+	// TODO: Add IgnoreResourceAttributeValue to scrapertest to ignore certain resource attributes
+	// node_name is dynamic based on MAC address, overwrite it with our expected name
+	for i := 0; i < actualMetrics.ResourceMetrics().Len(); i++ {
+		if _, ok := actualMetrics.ResourceMetrics().At(i).Resource().Attributes().Get("node_name"); ok {
+			actualMetrics.ResourceMetrics().At(i).Resource().Attributes().Remove("node_name")
+			actualMetrics.ResourceMetrics().At(i).Resource().Attributes().InsertString("node_name", "BB9020011AC4202")
+		}
+	}
```

Comment:
I think this was all meant to work around the problem solved by `IgnoreResourceAttributeValue`
```suggestion
```

Reply:
Agh, thought I removed that, sorry